dockstore-client/src/main/java/io/dockstore/client/cli/nested/AbstractEntryClient.java

```diff
@@ -1226,22 +1229,66 @@ public WesRequestData aggregateWesRequestData(final List<String> args) {
             out("Could not get WES section from config file");
         }
 
-        // Attempt to find the WES URI
-        final String wesUri = optVal(args, "--wes-url", null);
-        final String wesEndpointUrl = ObjectUtils.firstNonNull(wesUri, Objects.requireNonNull(configSubNode).getString("url"));
+        // Obtain the WES command object
+        JCommander parsedCommand = wesCommandParser.jCommander
+            .findCommandByAlias(wesCommandParser.jCommander.getParsedCommand());
+        WesCommandParser.WesMain command = parsedCommand == null ?  new WesCommandParser.WesMain() : (WesCommandParser.WesMain) parsedCommand.getObjects().get(0);
+
+        // Attempt to find the WES URL
+        final String wesEndpointUrl = ObjectUtils.firstNonNull(
+            command.getWesUrl(),
+            Objects.requireNonNull(configSubNode).getString("url"));
+
+        // Determine the authorization method used by the user
+        final String authType = ObjectUtils.firstNonNull(
+            command.getWesAuthType(),
+            Objects.requireNonNull(configSubNode).getString("type"));
+
+        // The auth value is either a bearer token or AWS profile
+        final String authValue = ObjectUtils.firstNonNull(
+            command.getWesAuthValue(),
+            Objects.requireNonNull(configSubNode).getString("authorization"));
 
         // Depending on the endpoint (AWS/non-AWS) we need to look for a different set of credentials
-        final boolean isAwsWes = flagPresent(args, "--aws");
+        final boolean isAwsWes = "aws".equals(authType);
         if (isAwsWes) {
-            // TODO should probably have the config sections all defined within a single class. At least have enums.
-            // AWS credentials shouldn't be stored in the dockstore config, so we wont check for it there.
-            final String accessKey = optVal(args, "--aws-access-key", null);
-            final String secretKey = optVal(args, "--aws-secret-key", null);
-            final String region = optVal(args, "--aws-region", null);
+
+            String accessKey = null;
+            String secretKey = null;
+
+            // If the user specified AWS as their intended target, but provided no profile name, assume they don't want to pass credentials
+            // in this request.
+            if (authValue != null) {
+
+                try {
+                    // Get the AWS config path
```

Comment:
I commented elsewhere, but I'm not convinced we need this. One could say it doesn't hurt since you already added it, but I do worry about all the command-line options.

Also, if I'm wrong and it's a fairly common practice, then perhaps it's fine.

Reply:
Removed. See comment above.